File: py/roundtrips/matching.py
from enum import Enum

class RoundtripMatching(Enum):
    """Enumerates algorithms used to match the offsetting order executions in a round-trip."""

    FIFO = 'fifo'
    """The offsetting order executions will be matched in FIFO (First In First Out) order."""

    LIFO = 'lifo'
    """The offsetting order executions will be matched in LIFO (Last In First Out) order."""

File: rs/src/roundtrips/matching.rs
/// Enumerates algorithms used to match the offsetting order executions
/// in a round-trip.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[repr(u8)]
pub enum RoundtripMatching {
    /// Matches offsetting order executions in First In First Out order.
    Fifo = 0,
    /// Matches offsetting order executions in Last In First Out order.
    Lifo = 1,
}
